myia/compile/backends/python/python.py

```diff
@@ -903,6 +1083,9 @@ def run(self, graph, backend):
         # reference: https://stackoverflow.com/a/19850183
         compiled = compile(final_code, "", "exec")
         module = ModuleType("mod")
+        # Add required global variables into module.
+        for name, value in self.globals.items():
+            module.__dict__[name] = value
         exec(compiled, module.__dict__)
         return getattr(module, "main")
```

Comment:
You don't need to create a module:

```suggestion
        exec(compiled, self.globals)
        return self.globals["main"]
```